synapse/lib/storm.py

```diff
@@ -574,7 +577,42 @@ def getArgParser(self):
             await snap.printf(f'...done')
             return
 
-        raise s_exc.SynErr('reindex was not told what to do!')
+        if self.opts.fire_handler:
+            obj = None
+            if self.opts.fire_handler.startswith('#'):
+                # Pull tag.... onadd handler???
+                name, _ = runt.snap.model.prop('syn:tag').type.norm(self.opts.fire_handler)
+                tname = '#' + name
+            else:
+                obj = runt.snap.model.prop(self.opts.fire_handler)
+                if obj is None:
+                    raise s_exc.NoSuchName(mesg='WORDS GO HERE',
+                                           name=self.opts.fire_handler)
+
+            async for node, path in genr:
+                if hasattr(obj, 'wasAdded'):
+                    if node.form.full != obj.full:
+                        continue
+                    await obj.wasAdded(node)
+                elif hasattr(obj, 'wasSet'):
+                    if obj.form.name != node.form.name:
+                        continue
+                    valu = node.get(obj.name)
+                    if valu is None:
+                        continue
+                    await obj.wasSet(node, valu)
+                else:
+                    # We're a tag...
+                    valu = node.get(tname)
+                    if valu is None:
+                        continue
+                    await runt.snap.core.runTagAdd(node, name, valu)
+
+                yield node, path
+
+            return
+
+        raise s_exc.SynErr(mesg='reindex was not told what to do!')
```

Comment:
suggest `One argument is required`. Actually, you could enforce this in the parser with `add_mutually_exclusive_group`. This would be a good idea because we don't currently have a check to see if more than one option is present.

Reply:
yeah - esp since this is a very surgical tool. now is a good time to change it since we can have multiple genr consumers.